node/src/components/storage.rs

```diff
@@ -327,7 +327,7 @@ impl Storage {
             // We use the opportunity for a small integrity check.
             assert_eq!(
                 *block.body_hash(),
-                block_body.hash(),
```

Comment:
Should we have a similar test with a protocol version belonging to the v2 hashes?

Reply:
I've actually had to change this, or it didn't work :+1: